### PR TITLE
Add a new config_stub fixture.

### PR DESCRIPTION
**Pull request by The-Compiler**
This replaces various other constructs:

- The `default_config` fixture - this means the config values used by `test_progress.py` are set explicitly and the (rather complex) default config code is mocked out.
- `stubs.ConfigStub` which was created by the tests manually before.

If you feel like I'd appreciate a quick review, @nicoddemus @hackebrot :smile:

### CODE REVIEW

**Comment by nicoddemus** [comment]
Looks good to me! :smile: 

:+1: +1

File: tests/conftest.py [hunk 1]
```diff
@@ -147,3 +150,49 @@ def cmdline_test(request):
     # Import qutebrowser.app so all cmdutils.register decorators get run.
     import qutebrowser.app  # pylint: disable=unused-variable
     return request.param
+
+
+class ConfigStub:
+
```
Comment (hackebrot) [inline tests/conftest.py]: Blank line :grinning:

File: tests/conftest.py [hunk 2]
```diff
@@ -147,3 +150,49 @@ def cmdline_test(request):
...
+        self.data = {}
+        self.changed = signal
+
+    def section(self, name):
+        """Get a section from the config.
+
+        Args:
+            name: The section name to get.
+
+        Return:
```
Comment (hackebrot) [inline tests/conftest.py]: I think this should be ``Returns:``, right?

http://sphinx-doc.org/latest/ext/example_google.html

**Comment by hackebrot** [comment]
:+1:

File: tests/conftest.py (hunk 1 above)
Comment (The-Compiler) [inline tests/conftest.py]: That's intentional - I'm not sure if [pep257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) (the document) says anything about it, but [pep257](https://pypi.python.org/pypi/pep257/) (the tool) does it this way:

`D20{3,4}	1 blank required {before,after} class docstring.`

Comment (hackebrot) [inline tests/conftest.py]: Fair enough :no_mouth:

File: tests/conftest.py (hunk 2 above)
Comment (The-Compiler) [inline tests/conftest.py]: Whoa, good point - the [Google styleguides](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Comments#Comments) say the same.

I'm using `Return:` (almost) everywhere in qutebrowser, but I also don't use Sphinx... Though adopting Sphinx will be the next thing after adopting pytest, and then I'll rewrite everything in that way.

**Pull request merged**